### ISSUE TITLE
Remove windows.h from public header?

### ISSUE THREAD

**Issue by derselbst** [t0rakka/mango#4]
Including `mango.hpp` includes `configure.hpp`, which includes `windows.h`. For me this causes several compilation errors, due to enum name clashes.

Currently, I could mitigate this by defining `WIN32_LEAN_AND_MEAN` before including `mango.hpp`. But I would prefer to move the `windows.h` into the cpp files. What do you think? There is also this line, not sure how to cope with that:

https://github.com/t0rakka/mango/blob/808b9a742e81044671542fc56a0fb81718f2ac1e/include/mango/window/window.hpp#L22

**Comment by t0rakka** [comment]
In theory could use the same type for NativeWindowHandle:  HWND is HANDLE, which is void* and then use <windows.h> internally indeed, but there are some side effects that I would like to avoid if possible with this.

The configure.hpp has the NOMINMAX defined, we could also define the WIN32_LEAN_AND_MEAN which would be "ideal" for my point of view, if that works with you we can go that way. If it doesn't, I'll take a better look at hiding the windows.h altogether.

**Comment by t0rakka** [comment]
One reason why I like the windows.h under "our" control is the NOMINMAX, without that the std::min and std::max become a pain in the butt and one has to mind this every time compiling for Windows and use std. Just one thing less to worry about, small platform specific details like this make coding less fun because every platform has their own quirks, like the X11 macros which are really "fun" -> keeping things clean systematically like this is just more pleasant. So the LEAN_AND_MEAN define would be ideal.

**Comment by derselbst** [comment]
Unfortunately, defining `WIN32_LEAN_AND_MEAN` in `configure.hpp` doesn't work, because then `DragQueryFileW`, `DragAcceptFiles` and `HDROP` are unknown which causes problems when compiling mango here:

https://github.com/t0rakka/mango/blob/808b9a742e81044671542fc56a0fb81718f2ac1e/source/mango/window/win32/win32_window.cpp#L536
https://github.com/t0rakka/mango/blob/808b9a742e81044671542fc56a0fb81718f2ac1e/source/mango/window/win32/win32_window.cpp#L649

**Comment by t0rakka** [comment]
Is the problem with Cygwin? I compiled with 32 bit and 64 bit VS2022 with WIN32_LEAN_AND_MEAN defined before including windows.h and everything compiles.

The HWND is a bit more tricky when look at it from the visual studio following the headers to actual declaration, the documentation lead to believe it's just void* (it isn't).

**Comment by t0rakka** [comment]
I went through some variations, but the cleanest is that define the WIN32_LEAN_AND_MEAN, then include the <shellapi.h> where it is needed (drag and drop).

The HWND compatibility wasn't as easy as void*, the documentation lead to believe that's how it works but when look into the actual headers the implementation has drifted to slightly different direction (it's pointer to struct with dummy int as member). I'd rather not change that too much, of course we could delay the windows.h appearance until we get to the window/GL/etc stuff.. but then the filesystem needs to include it by other means as it's using it as well. 

OK, so I did push the proposal to main.. I hope that does the trick because other options are a bit messy..
